### PR TITLE
Data generation script

### PR DESCRIPTION
**Pull request by abhogle**

### CODE REVIEW

File: .eslintignore [hunk 1]
```diff
@@ -1,2 +1,3 @@
 dist/*.js
-webpack.config.js
\ No newline at end of file
+webpack.config.js
+builds/
```
Comment (redperezz) [inline .eslintignore:3]: ✓

File: .gitignore [hunk 2]
```diff
@@ -2,3 +2,7 @@
 node_modules/
 dist/
 nohup.out
+seed/comments.csv
+seed/mongoimport.sh
+builds/
+cache/
```
Comment (redperezz) [inline .gitignore:8]: ✓

File: .gitlab-ci.yml [hunk 3]
```diff
@@ -25,7 +25,8 @@ test:
   stage: test
   before_script:
     - nohup node server/index.js &
-    - npm run seed
+    - npm run generate-data-test
+    - npm run seed-mongo-test
```
Comment (redperezz) [inline .gitlab-ci.yml:29]: ✓

File: controller/comments.js [hunk 4]
```diff
@@ -0,0 +1,177 @@
+/* eslint-disable camelcase */
+/* eslint-disable no-console */
+const db = require('../db');
+
+const allComments = async (req, res) => {
```
Comment (redperezz) [inline controller/comments.js:5]: ✓

File: controller/comments.js [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+    }
+
+    if (!comment) {
+      return res.status(400).json({
+        success: false,
+        msg: `no song with id ${id}`,
+      });
+    }
+
+    if (comment.length === 0) {
```
Comment (redperezz) [inline controller/comments.js:38]: Nice check!

File: controller/comments.js [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+  } catch (error) {
+    console.error(error);
+    res.status(400).json({
+      success: false,
+      msg: error,
+    });
+  }
+};
+
+const findSong = async (req, res) => {
```
Comment (redperezz) [inline controller/comments.js:22]: ✓

File: controller/comments.js [hunk 7]
```diff
@@ -0,0 +1,177 @@
...
+  } catch (error) {
+    console.error(error);
+    return res.status(400).json({
+      success: false,
+      msg: error,
+    });
+  }
+};
+
+const findComment = async (req, res) => {
```
Comment (redperezz) [inline controller/comments.js:58]: ✓

File: controller/comments.js [hunk 8]
```diff
@@ -0,0 +1,177 @@
...
+  } catch (error) {
+    console.error(error);
+    return res.status(400).json({
+      success: false,
+      msg: error,
+    });
+  }
+};
+
+const addComment = async (req, res) => {
```
Comment (redperezz) [inline controller/comments.js:94]: ✓

File: controller/comments.js [hunk 9]
```diff
@@ -0,0 +1,177 @@
...
+  } catch (error) {
+    console.error(error);
+    return res.status(400).json({
+      success: false,
+      msg: error,
+    });
+  }
+};
+
+const updateComment = async (req, res) => {
```
Comment (redperezz) [inline controller/comments.js:127]: ✓

File: controller/comments.js [hunk 10]
```diff
@@ -0,0 +1,177 @@
...
+  } catch (error) {
+    console.error(error);
+    return res.status(400).json({
+      success: false,
+      msg: error,
+    });
+  }
+};
+
+const deleteComment = async (req, res) => {
```
Comment (redperezz) [inline controller/comments.js:150]: ✓

File: controller/comments.js [hunk 11]
```diff
@@ -0,0 +1,177 @@
...
+};
+
+module.exports = {
+  allComments,
+  findSong,
+  findComment,
+  addComment,
+  updateComment,
+  deleteComment,
+};
```
Comment (redperezz) [inline controller/comments.js:177]: Nice job!

File: seed/mongoimport.js [hunk 12]
```diff
@@ -0,0 +1,16 @@
...
+  .then((csvData) => {
+    // eslint-disable-next-line no-console
+    db.dropDB()
+      .then(() => db.insertMany(csvData))
+      .then(() => {
+        setTimeout(() => {
+          mongoose.disconnect();
+        }, 1000);
+      });
+  });
```
Comment (redperezz) [inline seed/mongoimport.js:16]: Ooh I like your csv function, nicely done!

File: db/index.js [hunk 13]
```diff
@@ -61,6 +61,18 @@ const updateComment = (comment_id, content) => Comment.updateOne({ comment_id },
...
+// eslint-disable-next-line consistent-return
+const insertMany = (comments) => Comment.insertMany(comments, (err, docs) => {
+  if (err) {
+    console.log(err);
+    return err;
+  }
+  console.log(JSON.stringify(docs.length));
+  return (docs.length);
+});
+
```
Comment (redperezz) [inline db/index.js:75]: Nice!✓ maybe just remove lines 21-23.

File: server/routes.js [hunk 14]
```diff
@@ -0,0 +1,13 @@
...
+const router = express.Router();
+
+router.get('/comments', comments.allComments);
+router.get('/comments/:id', comments.findSong);
+router.get('/user/comment/:id', comments.findComment);
+router.post('/comment', comments.addComment);
+router.put('/comment/:id', comments.updateComment);
+router.delete('/comment/:id', comments.deleteComment);
+
+module.exports = router;
```
Comment (redperezz) [inline server/routes.js:13]: Well done!

File: server/index.js [hunk 15]
```diff
@@ -3,9 +3,10 @@ const express = require('express');
 const cors = require('cors');
 const path = require('path');
 const expressStaticGzip = require('express-static-gzip');
-const db = require('../db');
+// const db = require('../db');
+const router = require('./routes');
```
Comment (redperezz) [inline server/index.js:7]: Good job using Router!

File: seed/seed.js [hunk 16]
```diff
@@ -0,0 +1,53 @@
...
+const transformOpts = { objectMode: true };
+
+const json2csv = new Transform(opts, transformOpts);
+input.pipe(json2csv).pipe(output);
+output.on('finish', () => {
+  console.log('seeded');
+  // eslint-disable-next-line max-len
+  // shell.exec('mongoimport --type csv -d fec-soundcloud-comments -c comments --headerline --drop /Users/abhogle/Documents/RPT23/SDC/comments-service/seed/comments.csv');
+  // mongoose.disconnect();
+});
```
Comment (redperezz) [inline seed/seed.js:53]: Good seeding file maybe just remove the commented out parts like line 4, other than that good job!

**Review by redperezz — APPROVED** [review]
Well done!

File: db/index.js (hunk 13 above)
Comment (abhogle) [reply db/index.js:75]: Removed from the next branch. Good catch.

File: seed/seed.js (hunk 16 above)
Comment (abhogle) [reply seed/seed.js:53]: Removed commented out code from the next branch. Good catch.

**Pull request merged**